### PR TITLE
[Frontend/AST] Add `-interface-compiler-version` option to frontend/modules

### PR DESCRIPTION
**Pull request by xedin**
This option is going to be used to indicate what compiler was used
to build swift interface/module which is required for SE-0438.

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please test

**Review by artemcm — APPROVED** [review]

File: lib/Serialization/ModuleFileSharedCore.h [hunk 1]
```diff
@@ -103,6 +103,9 @@ class ModuleFileSharedCore {
   /// Name to use in public facing diagnostics and documentation.
   StringRef PublicModuleName;
 
+  /// The version of the Swift compiler used to build this module.
+  llvm::VersionTuple SwiftCompilerVersion;
```
[the remark this answers is not in the document]
Comment (xedin) [reply lib/Serialization/ModuleFileSharedCore.h:107]: Sure!

**Comment by xedin** [comment]
@swift-ci please test Linux platform

**Comment by xedin** [comment]
@swift-ci please test Windows platform

**Comment by xedin** [comment]
@swift-ci please test

**Comment by xedin** [comment]
@swift-ci please test

**Review by artemcm — APPROVED** [review]
LGTM!

**Pull request merged**